modules/gapi/include/opencv2/gapi/gkernel.hpp

```diff
@@ -30,6 +30,7 @@ struct GTypeInfo
 {
     GShape                 shape;
     cv::detail::OpaqueKind kind;
+    detail::HostCtor       ctor;
```

Comment:
Why do you need `ctor` here? I believe it's a property only for arrays and opaques

Reply:
It can be `optional<>` but is it really important ?

Reply:
I'm ok with `ctor` here. But please, add a comment that only arrays and opaques using it